lib/opam_cmd.ml

```diff
@@ -212,16 +212,17 @@ let get_opam_info ~root ~pins packages =
 let package_is_valid { package; dev_repo; _ } =
   match dev_repo with
   | `Error msg ->
-      R.error_msg (Fmt.strf "Do not know how to deal with %a: %s" pp_package package msg)
+      Logs.warn (fun l -> l "Do not know how to deal with %a: %s" pp_package package msg);
+      false
   | `Unknown msg ->
-      R.error_msg (Fmt.strf "Need a Duniverse fork for %a: %s" pp_package package msg)
-  | _ -> R.ok ()
+      Logs.warn (fun l -> l "Need a Duniverse fork for %a: %s" pp_package package msg);
+      false
+  | _ -> true
 
 let check_packages_are_valid pkgs =
```

Comment:
This function should probably be renamed to `filter_invalid_packages` or `filter_unsupported_packages`.

Reply:
Might actually be worth grouping all the package filtering together in a single function. That function could then be used in #23 to determine which package should be installed through `opam`.

Reply:
Renamed the function. Yes that would be a good refactor in the future.